### PR TITLE
provider/aws: Switch to session.Copy to build up configs

### PR DESCRIPTION
**Pull request by phinze**
Found this function which makes the config setup much nicer - looks like
it's meant to be used for the very thing we're doing, which is starting
w/ a base config and customizing it slightly.

https://docs.aws.amazon.com/sdk-for-go/api/aws/session/Session.html#Copy-instance_method

This will make the User-Agent setting code to follow much simpler.

### CODE REVIEW

File: builtin/providers/aws/config.go [hunk 1]
```diff
@@ -175,27 +172,14 @@ func (c *Config) Client() (interface{}, error) {
 		// signature format v4 requires region to be us-east-1 for global
 		// endpoints:
 		// http://docs.aws.amazon.com/general/latest/gr/sigv4_changes.html
-		usEast1AwsConfig := &aws.Config{
-			Credentials: creds,
-			Region:      aws.String("us-east-1"),
-			MaxRetries:  aws.Int(c.MaxRetries),
```
Comment (catsby) [inline builtin/providers/aws/config.go]: We seem to lose `c.MaxRetries` in this

**Comment by catsby** [comment]
Less code yay!

> which is starting w/ a base config and customizing it slightly.

I'm not seeing the "base config" here though, aren't we creating a new `&aws.Config` with each copy here?

```go
sess.Copy(&aws.Config{Endpoint: aws.String(c.DynamoDBEndpoint)})
```

It _seems_ like we're losing `c.MaxRetries`, and the `cleanhttp.DefaultClient()` addition here. Am I missing something?

**Comment by phinze** [comment]
@catsby base config we're copying is here:

https://github.com/hashicorp/terraform/pull/5623/files#diff-03b1112bc47923841f00d4153a75379cL145

File: builtin/providers/aws/config.go (hunk 1 above)
Comment (phinze) [inline builtin/providers/aws/config.go]: It's copied over from the `awsConfig` on line 145.

**Comment by catsby** [comment]
Ah yeah, my bad! I was thinking copying _configs_, not _sessions_ LGTM :+1:

**Pull request merged**